diff --git a/lib/jsdoc/util/dumper.js b/lib/jsdoc/util/dumper.js
index 515c9729453d7864abb49a4920e89dce660aa87b..396910a991bf70e10a2f37001946a9b8ba6cfcdf 100644
--- a/lib/jsdoc/util/dumper.js
+++ b/lib/jsdoc/util/dumper.js
@@ -95,13 +95,13 @@ class ObjectWalker {
                 return newArray;
             });
         }
-        else if ( util.isRegExp(o) ) {
+        else if ( Object.prototype.toString.call(o) === "[object RegExp]" ) {
             result = `<RegExp ${o}>`;
         }
-        else if ( util.isDate(o) ) {
+        else if ( Object.prototype.toString.call(o) === "[object Date]" ) {
             result = `<Date ${o.toUTCString()}>`;
         }
-        else if ( util.isError(o) ) {
+        else if ( Object.prototype.toString.call(o) === "[object Error]" ) {
             result = { message: o.message };
         }
         else if ( this.isFunction(o) ) {
